### PR TITLE
Add better `show` methods

### PR DESCRIPTION
**Pull request by nicoleepp**
Closes Issue #8 

Types updated for display were:
DeferredFuture
DeferredChannel

### CODE REVIEW

File: src/DeferredFutures.jl [hunk 1]
```diff
@@ -13,6 +13,15 @@ else
 end
 
 """
+    show(io::IO, ref::RemoteChannel)
+
+Print a simplified string representation of the `RemoteChannel` with its parameters.
+"""
+function Base.show(io::IO, ref::RemoteChannel)
```
Comment (iamed2) [inline src/DeferredFutures.jl]: Unfortunately you shouldn't do this. This overrides Base behaviour for Base types, which is called "type piracy". 

On the bright side, you are a pirate!

Comment (rofinn) [reply src/DeferredFutures.jl]: Yarr

Comment (nicoleepp) [reply src/DeferredFutures.jl]: Ah, ok! ☠️

**Comment by rofinn** [comment]
👍

**Review by rofinn — APPROVED** [review]

**Pull request merged**